app/models/order.rb

```diff
@@ -200,6 +199,12 @@ def asset_input_methods
     @asset_input_methods ||= DEFAULT_ASSET_INPUT_METHODS
   end
 
+  # request_type_ids_list is set for orders created by submission templates
+  # It is used by the input_field_infos section, which controls rendering
+  # form fields appropriate to each request type in the submission interface
+  # {request_type_ids} is calculated from this in the various sub-classes
+  # and gets persisted to the database, and used for the actual construction.
+  # TODO: Simplify this
```

Comment:
Is it worth putting down an idea of how to simplify this? For someone unfamiliar with the code it's not enough of a clue, I think

Reply:
I'll see what I can do, in truth the summary was a little vague as I had trouble keeping it all in my head. It might be by the time I've summarised it, I'll pretty much fixed it.